onnxruntime/core/mlas/lib/snchwc.cpp

```diff
@@ -201,21 +204,58 @@ Return Value:
             WorkBlock->Padding[dim + Dimensions] = 0;
         }
 
+        CanFlattenShape &= (WorkBlock->Padding[dim] == 0 && WorkBlock->Padding[dim + Dimensions] == 0);
+
         if (StrideShape != nullptr) {
             WorkBlock->StrideShape[dim] = size_t(StrideShape[dim]);
         } else {
             WorkBlock->StrideShape[dim] = 1;
         }
 
-        //
-        // Compute the number of output elements affected by left and right
-        // padding.
-        //
+        CanFlattenShape &= (WorkBlock->StrideShape[dim] == 1);
+    }
+
+    WorkBlock->InputSize = InputSize;
+    WorkBlock->OutputSize = OutputSize;
+
+    //
+    // Detect operations where the kernel is using the entire input width,
+    // has strides and dilations set to one, and no padding. These operations
```

Comment:
wondering why "strides" needs to be one please? thought it could be flatten as long as the input width (even after padding) is the same as kernel width.

Reply:
Yeah, stride could be >1 and I could then override the StrideShape[0] below. I was just keeping the pattern match simple to line up with what was done in the old convolution path. It's not something likely to happen in the wild.

Reply:
Things get complicated if stride_height > 1. It's easiest to just keep the code as I have it as it handles the models that matter. Is the code okay otherwise?